devito/data.py

```diff
@@ -122,11 +126,91 @@ def _alloc_C_libcall(self, size, ctype):
         alignment = self.lib.getpagesize()
         ret = self.lib.posix_memalign(ctypes.byref(c_pointer), alignment, c_bytesize)
         if ret == 0:
-            return c_pointer
+            return c_pointer, (c_pointer, )
         else:
-            return None
+            return None, None
 
-    def free(self, c_pointer, size):
+    def free(self, c_pointer):
+        self.lib.free(c_pointer)
+
+
+class GuardAllocator(MemoryAllocator):
+    """
+    Memory allocator based on ``posix`` functions. The allocated memory is
+    aligned to page boundaries.  Additionally, it allocates extra memory
+    before and after the data, and configures it so that an SEGV is thrown
+    immediately if an out-of-bounds access occurs.
+
+    Further, the remainder region of the last page (which cannot be
+    protected), is poisoned with NaNs.
+    """
+
+    is_Posix = True
+
+    @classmethod
+    def initialize(cls):
```

Comment:
If you inherit directly from ``PosixAllocator`` you won't need this class method, nor the ``is_Posix`` flag. Not sure if there are other implications though

Reply:
ah yes, good plan.  Will do that.